samanthas_telegram_bot/api_queries/person_creator.py

```diff
@@ -0,0 +1,166 @@
+                f"Created student [{user_data.first_name} {user_data.last_name}]"
+                f"({API_URL_STUDENT_RETRIEVE}{personal_info_id}), ID {personal_info_id}"
+            ),
+            failure_message=f"Failed to create student with ID {personal_info_id}",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.SUCCESS_AND_FAILURE,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+
+        if not user_data.student_assessment_answers:
+            logger.info(f"Chat {user_data.chat_id}: no assessment answers to send")
+            return result is not None
+
+        result = await send_to_backend(
+            context=context,
+            method=HttpMethod.POST,
+            url=API_URL_ENROLLMENT_TEST_SEND_RESULT,
+            data=DataMaker.student_enrollment_test(
+                personal_info_id=personal_info_id, user_data=user_data
+            ),
+            expected_status_code=httpx.codes.CREATED,
+            logger=logger,
+            success_message=f"Added assessment answers for {personal_info_id=}",
+            failure_message=f"Failed to send written assessment for {personal_info_id=})",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+        return result is not None
+
+    @classmethod
+    async def teacher(cls, update: Update, context: CUSTOM_CONTEXT_TYPES) -> bool:
+        """Sends a POST request to create an adult teacher."""
+
+        personal_info_id = await cls._personal_info_get_id(context)
+        user_data = context.user_data
+
+        result = await send_to_backend(
+            context=context,
+            method=HttpMethod.POST,
+            url=API_URL_TEACHERS_LIST_CREATE,
+            data=DataMaker.teacher(
+                update=update, personal_info_id=personal_info_id, user_data=user_data
+            ),
+            expected_status_code=httpx.codes.CREATED,
+            logger=logger,
+            success_message=(
+                f"Created adult teacher [{user_data.first_name} {user_data.last_name}]"
+                f"({API_URL_TEACHER_RETRIEVE}{personal_info_id}), ID {personal_info_id}"
+            ),
+            failure_message=f"Failed to create adult teacher with ID {personal_info_id}",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.SUCCESS_AND_FAILURE,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+        return result is not None
+
+    @classmethod
+    async def teacher_under_18(cls, update: Update, context: CUSTOM_CONTEXT_TYPES) -> bool:
+        """Sends a POST request to create a teacher under 18 years old."""
+
+        personal_info_id = await cls._personal_info_get_id(context)
+        user_data = context.user_data
+
+        result = await send_to_backend(
```

Comment:
It really will look better here if you just write `await client.create_teacher(teacher)`

Reply:
Done